src/backend/marsha/core/models/video.py

```diff
@@ -4,207 +4,26 @@
-    """
-    Model representing accesses to playlists that are granted to users.
-
-    ``through`` model between ``Playlist.users`` and ``User.playlists``.
-    """
-
-    # we allow deleting entries in this through table
-    _safedelete_policy = HARD_DELETE
-
-    user = models.ForeignKey(
-        to="User",
-        related_name="playlist_accesses",
-        verbose_name=_("user"),
-        help_text=_("user who has access to the playlist"),
-        # link is (soft-)deleted if user is (soft-)deleted
-        on_delete=models.CASCADE,
-    )
-    playlist = models.ForeignKey(
-        to="Playlist",
-        related_name="user_accesses",
-        verbose_name=_("playlist"),
-        help_text=_("playlist to which the user has access"),
-        # link is (soft-)deleted if playlist is (soft-)deleted
-        on_delete=models.CASCADE,
-    )
-    role = models.CharField(
-        max_length=20,
-        choices=ROLE_CHOICES,
-        verbose_name=_("role"),
-        help_text=_("role granted to the user on the consumer site"),
-        default=INSTRUCTOR,
-    )
-
-    class Meta:
-        """Options for the ``PlaylistAccess`` model."""
-
-        db_table = "playlist_access"
-        verbose_name = _("playlist access")
-        verbose_name_plural = _("playlist accesses")
-        constraints = [
-            models.UniqueConstraint(
-                fields=["user", "playlist"],
-                condition=models.Q(deleted=None),
-                name="playlist_access_unique_idx",
-            )
-        ]
-
-
-class Video(BaseModel):
+class Video(BaseFile):
     """Model representing a video, created by an author."""
 
-    title = models.CharField(
-        max_length=255, verbose_name=_("title"), help_text=_("title of the video")
-    )
     description = models.TextField(
         verbose_name=_("description"),
         help_text=_("description of the video"),
         null=True,
         blank=True,
     )
-    lti_id = models.CharField(
-        max_length=255,
-        verbose_name=_("lti id"),
-        help_text=_("ID for synchronization with an external LTI tool"),
-    )
-    created_by = models.ForeignKey(
-        to="User",
-        related_name="created_videos",
-        verbose_name=_("author"),
-        help_text=_("author of the video"),
-        # video is (soft-)deleted if author is (soft-)deleted
-        on_delete=models.CASCADE,
-        null=True,
-        blank=True,
-    )
-    playlist = models.ForeignKey(
-        to="Playlist",
-        related_name="videos",
-        verbose_name=_("playlist"),
-        help_text=_("playlist to which this video belongs"),
-        # don't allow hard deleting a playlist if it still contains videos
-        on_delete=models.PROTECT,
-    )
     position = models.PositiveIntegerField(
         verbose_name=_("position"),
         help_text=_("position of this video in the playlist"),
         default=0,
     )
-    duplicated_from = models.ForeignKey(
```

Comment:
The `description` and `position` fields should also be useful on File objects?

Reply:
In fact I don't really know. I don't understand why we have them, we don't use them.

Reply:
They are here with the GUI in mind... a youtube like admin in which you would see all your... per playlist... ranked and with a description eventually. That's why I think they also make sense on the Document object...

Reply:
In fact they are already in the `BaseFile` object.